scripts/west_commands/runners/blackmagicprobe.py

```diff
@@ -74,37 +119,45 @@ def check_call_ignore_sigint(self, command):
             signal.signal(signal.SIGINT, previous)
 
     def bmp_attach(self, command, **kwargs):
-        if self.elf_file is None:
-            command = (self.gdb +
-                       ['-ex', "set confirm off",
-                        '-ex', "target extended-remote {}".format(
-                            self.gdb_serial)] +
-                        self.connect_srst_disable_arg +
-                       ['-ex', "monitor swdp_scan",
-                        '-ex', "attach 1"])
+        if self.file is not None:
+            if self.file_type != FileType.ELF:
+                raise ValueError('Cannot debug; elf file required')
+            elf_name = self.file
+        elif self.elf_file is None:
+            raise ValueError('Cannot debug; elf is missing')
```

Comment:
This used to support just attaching, with no `file`; why is this feature being removed?